packages/axios-extension/src/abap/abap-service-provider.ts

```diff
@@ -49,9 +44,24 @@ export class AbapServiceProvider extends ServiceProvider implements AbapServiceP
      * @returns ABAP Transport Organizer settings
      */
     public async getAtoInfo(): Promise<AtoSettings> {
+        const serviceSchema = await this.getAdtCatalogService().getServiceDefinition(
+            AdtServiceConfigs[AdtServiceName.AtoSettings]
+        );
+
+        // Service not available on target ABAP backend version, return empty setting config
+        if (!serviceSchema) {
+            this.atoSettings = {};
+            return this.atoSettings;
+        }
+
         if (!this.atoSettings) {
             try {
-                const response = await this.get(ATO_CATALOG_URL_PATH);
+                const acceptHeaders = {
+                    headers: {
+                        Accept: 'application/*'
+                    }
+                };
+                const response = await this.get(serviceSchema.href, acceptHeaders);
```

Comment:
What's supposed to happen if `serviceSchema.href` is empty or `undefined`?

Reply:
@ullasholla  If serviceSchema is not empty, href should always be available.

Reply:
We're calling an external service and we have no control over its behavior (or bugs or misconfiguration). Calling `get()` later with an undefined/empty path may result in unpredictable and hard to debug behavior.

Reply:
Thanks @slin-sap. Looks good. One of the last things to do: can we please log an error/warning when we're returning nothing? This would be a case when there's a bug/misconfiguration in the backend. Logging would expedite trouble-shooting